app/javascript/utils/react-bootloader.jsx

```diff
@@ -24,27 +24,65 @@ Bugsnag.start({
 
 const ErrorBoundary = Bugsnag.getPlugin('react').createErrorBoundary(React)
 
+// This changes any extra things that need changing from the
+// turbo frame, such as body class or page title
+document.addEventListener('turbo:frame-render', (e) => {
+  const bodyClass = e.detail.fetchResponse.response.headers.get(
+    'exercism-body-class'
+  )
+  document.querySelector('body').className = bodyClass
+})
+
+/********************************************************/
+/** Add a loading cursor when a turbo-frame is loading **/
+/********************************************************/
+const setTurboStyle = (style) => {
+  const styleElemId = 'turbo-style'
+  if (!document.querySelector(`#${styleElemId}`)) {
+    const elem = document.createElement('style')
+    elem.id = styleElemId
+    document.head.appendChild(elem)
+  }
+
+  document.querySelector(`#${styleElemId}`).textContent = style
+}
+document.addEventListener('turbo:before-fetch-request', () => {
+  setTurboStyle('* { cursor:wait !important }')
+})
+document.addEventListener('turbo:before-render', () => {
+  setTurboStyle('')
+})
```

Comment:
You could make this one-liners:

```suggestion
document.addEventListener('turbo:before-fetch-request', () => setTurboStyle('* { cursor:wait !important }'))
document.addEventListener('turbo:before-render', () => setTurboStyle(''))
```

Reply:
This might be auto-formatted, but I think it is more readable in 3 lines than 1